backends/text-to-speech/macosx/macosx-text-to-speech.mm

```diff
@@ -112,12 +113,16 @@ - (void)ignoreNextFinishedSpeaking:(BOOL)ignore {
 #endif
 	}
 
-	_messageQueue.push(SpeechText(text, encoding));
+	_messageQueue.push(SpeechText(strToSpeak, encoding));
```

Comment:
That is incorrect.
You have
```
const Common::U32String &text
Common::String strToSpeak = text.encode();
```
So `strToSpeak` is UTF-8 and cannot be anything else.

But you also use
```
	if (encoding.empty()) {
#ifdef USE_TRANSLATION
		encoding = TransMan.getCurrentCharset();
#endif
	}
```
So UTF-32 encoding. There is a discrepancy between the two.

I think you can just remove the encoding field in `SpeechText`, and in `startNextSpeech` assume `kCFStringEncodingUTF8`.